src/main/java/org/gridsuite/modification/server/entities/equipment/modification/GeneratorModificationEntity.java

```diff
@@ -215,30 +216,30 @@ public void update(@NonNull ModificationInfos modificationInfos) {
-        this.energySource = new EnumModificationEmbedded<>(generatorModificationInfos.getEnergySource());
-        this.minActivePower = new DoubleModificationEmbedded(generatorModificationInfos.getMinActivePower());
-        this.maxActivePower = new DoubleModificationEmbedded(generatorModificationInfos.getMaxActivePower());
-        this.ratedNominalPower = new DoubleModificationEmbedded(generatorModificationInfos.getRatedNominalPower());
-        this.activePowerSetpoint = new DoubleModificationEmbedded(generatorModificationInfos.getActivePowerSetpoint());
-        this.reactivePowerSetpoint = new DoubleModificationEmbedded(generatorModificationInfos.getReactivePowerSetpoint());
-        this.voltageRegulationOn = new BooleanModificationEmbedded(generatorModificationInfos.getVoltageRegulationOn());
-        this.voltageSetpoint = new DoubleModificationEmbedded(generatorModificationInfos.getVoltageSetpoint());
-        this.plannedActivePowerSetPoint = new DoubleModificationEmbedded(generatorModificationInfos.getPlannedActivePowerSetPoint());
-        this.marginalCost = new DoubleModificationEmbedded(generatorModificationInfos.getMarginalCost());
-        this.plannedOutageRate = new DoubleModificationEmbedded(generatorModificationInfos.getPlannedOutageRate());
-        this.forcedOutageRate = new DoubleModificationEmbedded(generatorModificationInfos.getForcedOutageRate());
-        this.minimumReactivePower = new DoubleModificationEmbedded(generatorModificationInfos.getMinimumReactivePower());
-        this.maximumReactivePower = new DoubleModificationEmbedded(generatorModificationInfos.getMaximumReactivePower());
-        this.participate = new BooleanModificationEmbedded(generatorModificationInfos.getParticipate());
-        this.droop = new FloatModificationEmbedded(generatorModificationInfos.getDroop());
-        this.transientReactance = new DoubleModificationEmbedded(generatorModificationInfos.getTransientReactance());
-        this.stepUpTransformerReactance = new DoubleModificationEmbedded(generatorModificationInfos.getStepUpTransformerReactance());
-        this.voltageRegulationType = new EnumModificationEmbedded<>(generatorModificationInfos.getVoltageRegulationType());
-        this.regulatingTerminalId = new StringModificationEmbedded(generatorModificationInfos.getRegulatingTerminalId());
-        this.regulatingTerminalType = new StringModificationEmbedded(generatorModificationInfos.getRegulatingTerminalType());
-        this.regulatingTerminalVlId = new StringModificationEmbedded(generatorModificationInfos.getRegulatingTerminalVlId());
-        this.qPercent = new DoubleModificationEmbedded(generatorModificationInfos.getQPercent());
-        this.reactiveCapabilityCurve = new BooleanModificationEmbedded(generatorModificationInfos.getReactiveCapabilityCurve());
+        this.energySource = generatorModificationInfos.getEnergySource() != null ? new EnumModificationEmbedded<>(generatorModificationInfos.getEnergySource()) : null;
```

Comment:
Why do we need those new null checks ?

Reply:
no need to create all those embedded objects with null value and null opType inside ...